nlp/2021_03_18_pke_keyword_extraction/pke_keyword_extraction.ipynb

```diff
@@ -0,0 +1,503 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "Keyword Extraction Tip-of-the-week.ipynb",
+      "provenance": [],
+      "collapsed_sections": [
+        "E5fga3OkHom-",
+        "lSSEhYB2ICDh"
+      ]
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ne2ic1nmhs0n"
+      },
+      "source": [
+        "# Tip of the week: \"The KEYNG (read *king*) is dead, long live the KEYNG!\"\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "rvXj0fPGMXX9"
+      },
+      "source": [
+        "For a long time the go-to library for graph-based keyword extraction has been Gensim. However, the next major release—version 4.0 which is currently in beta—will [remove the entire summarisation module](https://github.com/RaRe-Technologies/gensim/wiki/Migrating-from-Gensim-3.x-to-4#12-removed-gensimsummarization), which includes the keyword extraction functionality.\n",
+        "\n",
+        "This motivates an update on **keyword extraction** from a previous Tip of the week on extractive summarization which included a section about keyword extraction with Gensim."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eQN6D4QVB2vM"
+      },
+      "source": [
+        "This notebook gives a brief overview of **pke** an **open-source keyphrase extraction toolkit**, that is easy to use, provides a wide range of keyword extraction methods which makes it easy to benchmark different approaches in order to choose the right algorithm for the problem at hand.\n",
+        "\n",
+        "In contrast to Gensim, which only provided keyword extraction using the widely known TextRank algorithm, `pke` offers [statistical models](https://boudinfl.github.io/pke/build/html/unsupervised.html#statistical-models), a [variety of TextRank flavours](https://boudinfl.github.io/pke/build/html/unsupervised.html#graph-based-models), as well as [simple supervised methods](https://boudinfl.github.io/pke/build/html/supervised.html) which even come pre-trained.\n",
+        "\n",
+        "In `pke` **preprocessing is built-in** using **spaCy**. This means the times of tormenting non-English languages with numerous preprocessing steps in order to make it look like English are finally over. *Bon vent !*\n",
+        "\n",
+        "Follow the code in this notebook to see how to use `pke` to extract keywords and a camparison of the extracted keywords.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "A_vPjpflJdkF"
+      },
+      "source": [
+        "---\n",
+        "\n",
+        "*My sincere apology to everyone who had to read this headline!*"
```

Comment:
why 😕 ?

Reply:
The headline sounded like a good idea at first but I grew more doubtful about it. 🤷

Reply:
Now I'm curious :D

Reply:
Doubtful about the pun or doubtful about the fact that we push pke as a replacement of gensim 🤔 ?

Reply:
Doubtful about the pun.

Reply:
The pun is great, no worries about it 😄 ! Would remove the 'sincere apology' line though to avoid confusion